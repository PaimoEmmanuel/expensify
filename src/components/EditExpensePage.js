import React from 'react';
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm';
import { editExpense } from '../actions/expenses'
import { startRemoveExpense } from '../actions/expenses';


export class EditExpensePage extends React.Component {

  onSubmit = (expense) => {
    this.props.editExpense(this.props.match.params.id, expense);
    this.props.history.push('/');
  }
  onRemove = () => {
      this.props.startRemoveExpense({ id: this.props.match.params.id });
      this.props.history.push('/');
  }
  render() {
    return (
      <div>
      Editing the expense with id of {this.props.match.params.id}
      <ExpenseForm 
      expense={this.props.expense}
      onSubmit = {this.onSubmit}/>
      <button onClick={this.onRemove}>Remove</button>
    </div>
    )
  }
}


const mapStateToProps = (state, props) =>{
  return {
    expense: state.expenses.find((expense => expense.id === props.match.params.id))
  }
}
const matchDispatchToProps = (dispatch, props) => {
  return {
    editExpense: (id, updates) => dispatch(editExpense(id, updates)),
    startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
  }
}
export default connect(mapStateToProps, matchDispatchToProps)(EditExpensePage);
