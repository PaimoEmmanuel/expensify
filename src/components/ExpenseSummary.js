import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = (props) => {
    console.log(props.expenseTotal);
    const expenseWord = props.expenseCount > 1 ? 'expenses' : 'expense';
     const formattedExpenseTotal = numeral(props.expenseTotal / 100).format('$0,0.00');
    return (

        <div>
            <p>Viewing {props.expenseCount} {expenseWord} totalling {formattedExpenseTotal}</p>
        </div>
    )
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return{
        expenseCount: visibleExpenses.length,
        expenseTotal: selectExpensesTotal(visibleExpenses) 
    }
};
export default connect(mapStateToProps)(ExpenseSummary);
