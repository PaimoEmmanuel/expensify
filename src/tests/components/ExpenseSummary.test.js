import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('Should render expense summary with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount ={1} expenseTotal ={4000}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should render expense summary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount ={4} expenseTotal ={444}/>);
    expect(wrapper).toMatchSnapshot();
});