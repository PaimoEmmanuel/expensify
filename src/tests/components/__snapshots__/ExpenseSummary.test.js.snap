// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense summary with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $40.00
  </p>
</div>
`;

exports[`Should render expense summary with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    4
     
    expenses
     totalling 
    $4.44
  </p>
</div>
`;
